app/src/main/java/mollie/tictactoe/MainActivity.java

```diff
@@ -14,53 +15,89 @@
 
 public class MainActivity extends AppCompatActivity {
 
-    private final List<Button> mBoard = new ArrayList<>();
+    private final String LOG_TAG = getClass().getName();
+    private List<Button> mBoard;
     private GameEngine mGameEngine;
-    private static final String EXTRA_PLAYER_TYPES = "tictactoe.game_type";
-    private static final String EXTRA_BOARD_TYPE = "tictactoe.board_type";
+    public static final String EXTRA_PLAYER_TYPES = "mollie.tictactoe.player_types";
+    public static final String EXTRA_BOARD_TYPE = "mollie.tictactoe.board_type";
+    public static final String EXTRA_GAME_TYPE = "mollie.tictactoe.game_type";
+    public static final String EXTRA_COMPUTER_FIRST = "mollie.tictactoe.computer_first";
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
+        mGameEngine = getGame();
+        mBoard = createBoard();
+        setClickable(isAComputerGame(EXTRA_GAME_TYPE));
+        checkComputerMoveFirst();
+    }
+
+    private void checkComputerMoveFirst() {
+        if (isAComputerGame(EXTRA_COMPUTER_FIRST)) {
+            playComputerMove();
+        }
+    }
+
+    private boolean isAComputerGame(String extraGameType) {
+        return getIntent().getBooleanExtra(extraGameType, false);
+    }
+
+    private GameEngine getGame() {
         String[] playersTypes = getIntent().getStringArrayExtra(EXTRA_PLAYER_TYPES);
-        int isA3x3Board = getIntent().getIntExtra(EXTRA_BOARD_TYPE, 3);
-        createGame(isA3x3Board);
-        createBoard();
-        setClickable();
+        int boardSize = getIntent().getIntExtra(EXTRA_BOARD_TYPE, 3);
+        return createGame(playersTypes, boardSize);
+    }
+
+    private void playComputerMove() {
+        int move = getComputerMove();
+        mGameEngine.play(move);
+        mBoard.get(move).setText(mGameEngine.board(move).toString());
+    }
+
+    private int getComputerMove() {
+        Integer move = null;
```

Comment:
Can `int` be used here instead of boxing then auto-boxing back to return an int?